samples/stock/demo/drag-on-axis/demo.js

```diff
@@ -0,0 +1,132 @@
+(function (Highcharts) {
+    // Configurable
+    const MOUSE_SENSITIVITY = 0.002,
+        SCROLL_SENSITIVITY = 0.0002;
+    function centerZoom(axis, strength) {
+        let zoomOffset = (axis.max - axis.min) / 2;
+        const center = axis.min + zoomOffset;
+
+        zoomOffset += strength * axis.toValue(1);
+
+        if (zoomOffset < 0) {
+            zoomOffset = 0;
+        }
+
+        axis.setExtremes(center - zoomOffset, center + zoomOffset, true, false);
+        if (axis.chart.resetZoomButton) {
+            axis.chart.resetZoomButton = axis.chart.resetZoomButton.destroy();
+        }
+        axis.chart.showResetZoom();
+    }
+
+    Highcharts.addEvent(Highcharts.Axis, 'afterInit', function () {
+        const axis = this,
+            renderer = axis.chart.renderer;
+
+        if (!axis.isXAxis && !axis.options.isInternal) {
```

Comment:
Probably wise to also check for `!this.chart.inverted` && `!this.chart.polar` just to be on the safe side.